python/prohibition_web_svc/middleware/event_middleware.py

```diff
@@ -0,0 +1,207 @@
+                linkage_other=data.get('linkage_other'),
+                speed_limit=data.get('speed_limit'),
+                vehicle_speed=data.get('vehicle_speed'),
+                speed_estimation_technique=data.get('speed_estimation_technique'),
+                speed_confirmation_technique=data.get('speed_confirmation_technique'),
+                incident_details=data.get('incident_details'),
+                incident_details_extra_page=data.get('incident_details_extra_page'),
+                created_dt=date_created,
+                updated_dt=date_created,
+            )
+            event.vi_form = vi_form   
+        if data.get('TwentyFourHour'):
+            return
+        if data.get('TwelveHour'):
+            return    
+        if data.get('IRP'):
+            return 
+        logging.debug('Saving Event')   
+        db.session.add(event)
+        db.session.commit()
+    except Exception as e:
+        logging.warning(str(e))
+        return False, kwargs
+    kwargs['response_dict'] = jsonify(event)
+    kwargs['event'] = event
+    return True, kwargs
+
+def save_event_pdf(**kwargs) -> tuple:
+    date_created=datetime.now()
+    logging.debug('save event pdf')
+    try:
+        data = kwargs.get('payload')
+        event = kwargs.get('event')
+        client = Minio(
+            Config.MINIO_BUCKET_URL,
+            access_key=Config.MINIO_AK,
+            secret_key=Config.MINIO_SK,
+            secure=Config.MINIO_SECURE,
+        )
+
+        # Make 'asiatrip' bucket if not exist.
+        found = client.bucket_exists("test")
+        if not found:
+            client.make_bucket("test")
+        else:
+            print("Bucket 'test' already exists")
+
+        if(data.get('VI')):
+            filename = str(uuid.uuid4().hex)            
+            pdf_filename = f"/tmp/{filename}.pdf"
+            encrypted_pdf_filename = f"/tmp/{filename}_encrypted.pdf"
+            b64encoded=data.get("VI_form_png").split(",")[1]
+            with open(f"/tmp/{filename}.png", "wb") as fh:
+                fh.write(b64decode(b64encoded))
+            pdf_bytes = img2pdf.convert(f"/tmp/{filename}.png")
+            with open(pdf_filename, "wb") as file:
+                file.write(pdf_bytes)
+            encryptPdf_method1(pdf_filename, Config.ENCRYPT_KEY, encrypted_pdf_filename)  
+            logging.debug('File encrypted')
+            encoded_file_name = f"{filename}_encrypted.pdf"
+            encoded_pdf_filepath = f'/tmp/{encoded_file_name}'
+            with open(encoded_pdf_filepath, 'rb') as file_data:
+                client.fput_object(Config.STORAGE_BUCKET_NAME, encoded_file_name, encoded_pdf_filepath)
+            logging.debug('File uploaded')
+            
+            form_storage = FormStorageRefs(
+                form_id_vi=event.vi_form.form_id,
+                event_id=event.event_id,
+                form_type='VI',
+                storage_key=f'test/{encoded_file_name}',
```

Comment:
the folder name should be read from an env variable. There is already an env variable passed on line 160. Thats the folder name (bucket name)

Reply:
changed to be the same as the config bucket